### ISSUE TITLE
Wide Class Table Forcing Empty Space, Crowding Further Columns

### ISSUE THREAD

**Issue by Waistel** [naturalcrit/homebrewery#444]
Instead of the columns building from left to right and adjusting width based on their content, for some reason the table seems to believe the text in the "Features" column takes up far much more room than it actually does. This leaves a large empty space at the end of the Features column, which in turn bunches up the spell slots in the class table. Also the titles for "Cantips known" and "Spells Known" in tables seem to be wider than necessary. I have tried everything, and the only conclusion I have reached is that the way the table accounts for text space is a bit messed up.

Is there a way to adjust this?

### Additional Details

Here's an example based on the UA Artificer table I was reproducing for a larger compilation.

**Brew code to reproduce** : <details><summary>Click to expand</summary><code><pre>

<div class='classTable wide'>
##### The Artificer
| Level | Proficiency Bonus | Features                                   | Spells Known | 1st | 2nd | 3rd | 4th |
|:-----:|:-----------------:|:-------------------------------------------|:------------:|:---:|:---:|:---:|:---:|
| 1st   | +2                | Artificer Specialist, Magic Item Analysis  | —            | —   | —   | —   | —   |
| 2nd   | +2                | Tool Expertise, Wondrous Invention         | —            | —   | —   | —   | —   |
| 3rd   | +2                | Artificer Specialist feature, Spellcasting | 3            | 2   | —   | —   | —   |
| 4th   | +2                | Ability Score Improvement, Infuse Magic    | 4            | 3   | —   | —   | —   |
| 5th   | +3                | Superior Attunement, Wondrous Invention    | 4            | 4   | —   | —   | —   |
| 6th   | +3                | Mechanical Servant                         | 4            | 3   | —   | —   | —   |
| 7th   | +3                | —                                          | 5            | 4   | 2   | —   | —   |
| 8th   | +3                | Ability Score Improvement                  | 5            | 4   | 2   | —   | —   |
| 9th   | +4                | Artificer Specialist feature               | 6            | 4   | 2   | —   | —   |
| 10th  | +4                | Woundrous Invention                        | 7            | 4   | 3   | —   | —   |
| 11th  | +4                | —                                          | 8            | 4   | 3   | —   | —   |
| 12th  | +4                | Ability Score Improvement                  | 8            | 4   | 3   | —   | —   |
| 13th  | +5                | —                                          | 9            | 4   | 3   | 2   | —   |
| 14th  | +5                | Artificer Specialist feature               | 10           | 4   | 3   | 2   | —   |
| 15th  | +5                | Superior Attunement, Woundrous Invention   | 10           | 4   | 3   | 2   | —   |
| 16th  | +5                | Ability Score Improvement                  | 11           | 4   | 3   | 3   | —   |
| 17th  | +6                | Artificer Specialise feature               | 11           | 4   | 3   | 3   | —   |
| 18th  | +6                | Ability Score Improvement                  | 11           | 4   | 3   | 3   | —   |
| 19th  | +6                | —                                          | 12           | 4   | 3   | 3   | 1   |
| 20th  | +6                | Soul of Artifice, Woundrous Invention      | 13           | 4   | 3   | 3   | 1   |
</div>
</pre></code></details>

**Comment by AlexeySachkov** [comment]
@Waistel, column width can be adjusted, but you will have to use some CSS. Here is a complete example, where "Features" column is set to have width of 10% of the table width: [share link](https://homebrewery.naturalcrit.com/share/msiongBI9oXz)

The source for it looks like:
```
<style>
#custom-table-wrapper table thead th:nth-child(3) {
	width: 10%;
}
</style>

##### The Artificer
<div class='wide' id='custom-table-wrapper'>
| Level | Proficiency Bonus | Features                                   | Spells Known | 1st | 2nd | 3rd | 4th |
|:-----:|:-----------------:|:-------------------------------------------|:------------:|:---:|:---:|:---:|:---:|
| 1st   | +2                | Artificer Specialist, Magic Item Analysis  | —            | —   | —   | —   | —   |
| 2nd   | +2                | Tool Expertise, Wondrous Invention         | —            | —   | —   | —   | —   |
</div>
```

Some explanation: I've added an `id` to the `div`, which contains the table, in order to uniquely distinguish that table from any others you have. Using that `id`, I've created a CSS rule, which targets 3rd `th` element (that corresponds to the header cell of the table containing "Features" text) and sets its width to be 10%. My understanding is that is 10% of the table width. You can put some absolute value here like `200px`, for example.

Please let us know if you have any further questions.

**Comment by calculuschild** [comment]
This is also something that can be solved with #1528
